components/match2/commons/starcraft_starcraft2/opponent_display_starcraft.lua

```diff
@@ -77,6 +78,8 @@ determined by its layout context, and not of the opponent.
 function StarcraftOpponentDisplay.BlockOpponent(props)
 	DisplayUtil.assertPropTypes(props, StarcraftOpponentDisplay.propTypes.BlockOpponent)
 	local opponent = props.opponent
+	-- Default TBDs to not show links
+	props.showLink = Logic.nilOr(props.showLink, not StarcraftOpponent.isTbd(opponent))
```

Comment:
Better to localize the value, since you are now modifying the input table (which will have it modified outside this function).

Reply:
the value needs to be passed along to further functions (in commons OpponentDisplay)

Reply:
Yes, but it's generally a good idea to avoid mutating inputs in display, as that input may to sent to one of more functions afterwards.

Reply:
so you want to have a local in line 82
then use that in line 88 and have a table merge in line 97 and 99 so we can pass it along there?
```lua
	-- Default TBDs to not show links
	local showLink = Logic.nilOr(props.showLink, not StarcraftOpponent.isTbd(opponent))

	if opponent.type == 'team' then
		return StarcraftOpponentDisplay.BlockTeamContainer({
			flip = props.flip,
			overflow = props.overflow,
			showLink = showLink,
			style = props.teamStyle,
			team = opponent.team,
			template = opponent.template or 'tbd',
		})
	elseif opponent.type == 'literal' and opponent.extradata.hasRaceOrFlag then
		props.showRace = false
		return StarcraftOpponentDisplay.PlayerBlockOpponent(
			Table.merge(props, {showLink = showLink})
		) -- here
	elseif opponent.type == 'literal' then
		return OpponentDisplay.BlockOpponent(props)
	else -- opponent.type == 'solo' 'duo' 'trio' 'quad'
		return StarcraftOpponentDisplay.PlayerBlockOpponent(
			Table.merge(props, {showLink = showLink})
		) -- and here
```

Reply:
Yup

Reply:
done